Content.Server/GameObjects/Components/Kitchen/KitchenMicrowaveComponent.cs

```diff
@@ -0,0 +1,416 @@
+using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.GameObjects.Components.Nutrition;
+using Content.Shared.Chemistry;
+using Robust.Shared.Serialization;
+using Robust.Shared.Interfaces.GameObjects;
+using Content.Shared.Prototypes.Kitchen;
+using Content.Shared.Kitchen;
+using Robust.Shared.Timers;
+using Robust.Server.GameObjects;
+using Content.Shared.GameObjects.Components.Power;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.GameObjects.Components.Container;
+using Content.Server.GameObjects.Components.Power;
+using Robust.Server.GameObjects.Components.UserInterface;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Localization;
+using Content.Server.Interfaces;
+using Robust.Shared.Audio;
+using YamlDotNet.Serialization.NodeTypeResolvers;
+
+namespace Content.Server.GameObjects.Components.Kitchen
+{
+    [RegisterComponent]
+    [ComponentReference(typeof(IActivate))]
+    public class KitchenMicrowaveComponent : SharedMicrowaveComponent, IActivate, IAttackBy, ISolutionChange
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager;
+        [Dependency] private readonly IEntityManager _entityManager;
+        [Dependency] private readonly RecipeManager _recipeManager;
+        [Dependency] private readonly IServerNotifyManager _notifyManager;
+#pragma warning restore 649
+
+#region YAMLSERIALIZE
+        private int _cookTimeDefault;
+        private int _cookTimeMultiplier; //For upgrades and stuff I guess?
+        private string _badRecipeName;
+        private string _startCookingSound;
+        private string _cookingCompleteSound;
+#endregion
+
+#region VIEWVARIABLES
+        [ViewVariables]
+        private SolutionComponent _solution;
+
+        [ViewVariables]
+        private bool _busy = false;
+
+        /// <summary>
+        /// This is a fixed offset of 5.
+        /// The cook times for all recipes should be divisible by 5,with a minimum of 1 second.
+        /// For right now, I don't think any recipe cook time should be greater than 60 seconds.
+        /// </summary>
+        [ViewVariables]
+        private uint _currentCookTimerTime { get; set; } = 1;
+#endregion
+
+        private bool Powered => _powerDevice.Powered;
+
+        private bool HasContents => _solution.ReagentList.Count > 0 || _storage.ContainedEntities.Count > 0;
+
+        void ISolutionChange.SolutionChanged(SolutionChangeEventArgs eventArgs) => UpdateUserInterface();
+
+        private AudioSystem _audioSystem;
+
+        private AppearanceComponent _appearance;
+        private PowerDeviceComponent _powerDevice;
+
+        private BoundUserInterface _userInterface;
+
+        private Container _storage;
+        /// <summary>
+        /// A dictionary of PrototypeIDs and integers representing quantity.
+        /// </summary>
+        private Dictionary<string, int> _solids;
```

Comment:
Doesn't make much sense to have this as a field when it's only populated by `wzhzhzh` and then subsequently only used by `CanSatisfyRecipe`, which is called only by `wzhzhzh`. Best to just make it a local var and pass it via a parameter.